home/templatetags/generic_components.py

```diff
@@ -9,24 +9,28 @@
 
 
 @register.inclusion_tag('generic_components/primary_button.html')
-def primary_button(title, extra_classnames='', href=None, icon_path=None,
+def primary_button(title, extra_classnames='', href=None, icon=None, icon_path=None,
                    font_color=None, background_color=None, is_svg_icon=False):
-    theme_settings = ThemeSettings.for_site(Site.objects.filter(is_default_site=True).first())
-
-    font_color = font_color or theme_settings.primary_button_font_color
-    background_color = background_color or theme_settings.primary_button_background_color
-
-    if icon_path and is_svg_icon and font_color:
-        icon_path = svg_to_png_url(icon_path, fill_color=font_color)
-
-    return {
-        'title': title,
-        'href': href,
-        'icon_path': icon_path,
-        'font_color': font_color,
-        'background_color': background_color,
-        'extra_classnames': extra_classnames,
-    }
+    try:
+        theme_settings = ThemeSettings.for_site(Site.objects.filter(is_default_site=True).first())
+
+        font_color = font_color or theme_settings.primary_button_font_color
+        background_color = background_color or theme_settings.primary_button_background_color
+
+        if (icon or icon_path) and is_svg_icon and font_color:
+            relative_path = f"media/{icon.file}" if icon else f"static/{icon_path}"
+            icon_path = svg_to_png_url(relative_path, fill_color=font_color)
+
+        return {
+            'title': title,
+            'href': href,
+            'icon_path': icon_path,
+            'font_color': font_color,
+            'background_color': background_color,
+            'extra_classnames': extra_classnames,
+        }
+    except:
+        return {}
```

Comment:
I think this except block is too generic. This will catch all the exceptions and we probably don't want that

Reply:
I admit that this error handling needs some more work. My rationale for this is that we would not want a missing SVG file, for example, to cause an internal server error. I would rather catch any (or most) errors, log it and move on. If that means there would be a missing button on the front-end, so be it.